### PR TITLE
Fixes a Transform Bug in Loop Fusion

### PR DESCRIPTION
**Pull request by sppalkia**
Also adds better error reporting for missing symbols during `uniquify`

### CODE REVIEW

File: weld/ast.rs [hunk 1]
```diff
@@ -523,21 +522,22 @@ impl<T: TypeBounds> Expr<T> {
 
     /// Recursively transforms an expression in place by running a function on it and optionally replacing it with another expression.
     pub fn transform_and_continue<F>(&mut self, func: &mut F)
-        where F: FnMut(&mut Expr<T>) -> Option<(Expr<T>, bool)>
+        where F: FnMut(&mut Expr<T>) -> (Option<Expr<T>>, bool)
```
Comment (mateiz) [inline weld/ast.rs]: What's the motivation for this change? Is the bool about whether to continue recursing down or something?

File: weld/ast.rs [hunk 2]
```diff
@@ -523,21 +522,22 @@ impl<T: TypeBounds> Expr<T> {
...
+            (Some(e), false) => {
                 *self = e;
             }
-            None => {
+            (None, true) => {
                 for c in self.children_mut() {
                     c.transform_and_continue(func);
                 }
             }
+            (None, false) => {}
```
Comment (mateiz) [inline weld/ast.rs]: This whole switch statement is kind of weird now; why don't we have an if on the first part of the tuple (replace `*self` or not) and then a second if on the second part (transform children)?

File: weld/ast.rs (hunk 1 above)
Comment (sppalkia) [inline weld/ast.rs]: Yup, the `bool` states whether to recurse down and the expression is what to replace it with. Before, erroring out was hard since there was no way of saying "don't replace the expression and don't recurse down."

File: weld/ast.rs (hunk 2 above)
Comment (sppalkia) [inline weld/ast.rs]: Hm, the reason for this is that if you do end up replacing self, you want to call the transform on yourself (to match on the replaced expression); otherwise, you call the transform on each child.

Comment (mateiz) [inline weld/ast.rs]: OK, that makes sense then.

**Comment by sppalkia** [comment]
Cool, merged this.

**Comment by deepakn94** [comment]
Hmm -- I think this commit broke some of the Pandas workloads. (seeing a segfault when running `python get_population_stats_simplified_grizzly.py`)
